redis/src/cluster_async/mod.rs

```diff
@@ -426,11 +425,14 @@ where
                         });
                         self.poll(cx)
                     }
-                    ErrorKind::IoError => Next::Reconnect {
-                        request: this.request.take().unwrap(),
-                        target: identifier,
+                    ErrorKind::IoError => {
+                        warn!("disconnected with err: {}", err);
```

Comment:
We shall capture the target node' address and IP too, e.g. node localhost:6379 (127.0.0.1) disconnected with... 
something like - 
```
let address = address_for_identifier(identifier);
let ip = ip_for_identifier(identifier);
("node {:?} ({:?}) disconnected with..." address, ip),

Reply:
Also can you add debug/trace logs for the different errors - "Handling error: received MOVED error: {}", ask, tryagain, clusterdown etc. It will help us to understand the internal handling

Reply:
we already trace the error at the top of the function.

> We shall capture the target node' address and IP too

this data isn't available here. notice that this function is on the `Request` type.

Reply:
It's important that we log this information, debugging issues without understanding which nodes raised errors doesn't give the required info. We won't be able to help customers to understand issues.
Is it possible to pass a pointer to the connection container, or to pass add this info to request?

Reply:
We already need this info - to understand the downtime we see in Polyglot in different workflows

Reply:
> Is it possible to pass a pointer to the connection container

Making the design more complex and less compartmentalized just to have better logs won't help us solve bugs, it will cause bugs down the road.

> or to add this info to request?

Sure, but that will increase the size of the `Request` object, and will increase memory churn. These things have costs, and you have to weigh the cost of the change by its benefits.

Have you tried using the current branch with Polyglot, and seen that the logs are insufficient?

Reply:
let's go over a polyglot run with the current outputs, and see what info is actually missing.

Reply:
I worked on 30453508 tickets, logs - like in lettuce - where they capture the node's address & IP, were the most helpful ones. Seeing that a moved error is thrown but not knowing who raised it makes it really hard to understand server/client issues. I think that as a project that emphasize support and being able to handle cluster failures we have to have logs that can give relevant information.